### PR TITLE
Add options for filtering commit list and subscribe by commit type, ignore alias commits by default

### LINKED ISSUE #6366: List commit shows alias commits
When running `pachctl list commit repo@branch` the returned value has the alias commits listed as well. This is confusing to the user, because (in the case of the Intermediate Housing Prices example) we have a data repository where we added one file and after everything runs there are 4 commits. 3 of these are alias commits and are confusing to the end user - 4 commits for one file. 

A couple of options: 
1. Filter out alias commits by default. 
2. Display a column that identifies the commit as user

### PR DESCRIPTION
**Pull request by Tryneus**
Fixes #6366.

This adds the fields `All` and `OriginKind` to the `ListCommitRequest` and `SubscribeCommitRequest` protobufs, and the corresponding options `--all` and `--origin=(user|alias|auto|fsck)` to `pachctl list commit` and `pachctl subscribe commit`.

### CODE REVIEW

File: src/pfs/pfs.proto [hunk 1]
```diff
@@ -239,6 +241,8 @@ message SubscribeCommitRequest {
   Commit from = 3;
   // Don't return commits until they're in (at least) the desired state.
   CommitState state = 4;
+  bool all = 5; // Return commits of all kinds (without this, aliases are excluded)
```
Comment (PFedak) [inline src/pfs/pfs.proto:244]: I guess it would be overkill to make these a shared subproto

File: src/internal/clientsdk/pfs.go [hunk 2]
```diff
@@ -85,3 +85,52 @@ func ForEachCommitSet(client pfs.API_ListCommitSetClient, cb func(*pfs.CommitSet
 	}
 	return nil
 }
+
```
Comment (PFedak) [inline src/internal/clientsdk/pfs.go:88]: ah nice, missed that this file was created

File: src/server/pfs/cmds/cmds.go [hunk 3]
```diff
@@ -442,27 +442,50 @@ $ {{alias}} foo@master --from XXX`,
...
 				encoder := cmdutil.Encoder(output, os.Stdout)
-				return c.ListCommitF(branch.Repo, toCommit, fromCommit, number, false, func(ci *pfs.CommitInfo) error {
+				return clientsdk.ForEachCommit(listClient, func(ci *pfs.CommitInfo) error {
 					return encoder.EncodeProto(ci)
 				})
 			} else if output != "" {
 				return errors.New("cannot set --output (-o) without --raw")
 			}
 			writer := tabwriter.NewWriter(os.Stdout, pretty.CommitHeader)
-			if err := c.ListCommitF(branch.Repo, toCommit, fromCommit, number, false, func(ci *pfs.CommitInfo) error {
```
Comment (PFedak) [inline src/server/pfs/cmds/cmds.go:454]: have we talked about removing ListCommitF entirely? feels a little late to do that, I guess

File: src/server/pfs/cmds/cmds.go [hunk 4]
```diff
@@ -535,12 +558,41 @@ $ {{alias}} test@master --new`,
 			}
 			defer c.Close()
 
+			var fromCommit *pfs.Commit
 			if newCommits && from != "" {
 				return errors.Errorf("--new and --from cannot be used together")
+			} else if newCommits || from != "" {
+				fromCommit = branch.NewCommit(from)
```
Comment (PFedak) [inline src/server/pfs/cmds/cmds.go:565]: I was going to comment that this was confusing in the case that branch isn't specified, but, as we discussed, `from` seems somewhat broken, so I'm not sure it matters

**Review by PFedak — APPROVED** [review]
LGTM, comments are just stream of consciousness

File: src/pfs/pfs.proto (hunk 1 above)
Comment (Tryneus) [reply src/pfs/pfs.proto:244]: Yeah, nested protos aren't very nice to use.

**Pull request merged**